stratum/hal/lib/common/p4_service.cc

```diff
@@ -210,57 +210,61 @@ ::grpc::Status ToGrpcStatus(const ::util::Status& status,
 
 // Helper to facilitate logging the write requests to the desired log file.
 void LogWriteRequest(uint64 node_id, const ::p4::v1::WriteRequest& req,
+                     const ::util::Status& status,
                      const std::vector<::util::Status>& results,
                      const absl::Time timestamp) {
   if (FLAGS_write_req_log_file.empty()) {
     return;
   }
-  if (results.size() != req.updates_size()) {
+  if (!results.empty() && results.size() != req.updates_size()) {
     LOG(ERROR) << "Size mismatch: " << results.size()
                << " != " << req.updates_size() << ". Did not log anything!";
     return;
   }
   std::string msg = "";
   std::string ts =
       absl::FormatTime("%Y-%m-%d %H:%M:%E6S", timestamp, absl::LocalTimeZone());
-  for (size_t i = 0; i < results.size(); ++i) {
-    absl::StrAppend(&msg, ts, ";", node_id, ";",
-                    req.updates(i).ShortDebugString(), ";",
-                    results[i].error_message(), "\n");
+  for (size_t i = 0; i < req.updates_size(); ++i) {
+    absl::StrAppend(
+        &msg, ts, ";", node_id, ";", req.updates(i).ShortDebugString(), ";",
+        results.empty() ? status.error_message() : results[i].error_message(),
+        "\n");
```

Comment:
If this change is to complicated, I'd propose to drop logging of write errors without details completely. Even today certain requests with errors (auth, valid device ID, ...) are not written to the log.